src/common/src/types/mod.rs

```diff
@@ -693,6 +699,103 @@ impl ScalarImpl {
+            ScalarImpl::List(_) => todo!(),
+        };
+        body
+    }
+
+    pub fn bytes_to_scalar(b: &Vec<u8>, data_type: &ProstDataType) -> Result<Self> {
+        let value = match data_type.get_type_name()? {
+            TypeName::Boolean => ScalarImpl::Bool(
+                i8::from_be_bytes(b.as_slice().try_into().map_err(|e| {
+                    InternalError(format!("Failed to deserialize bool, reason: {:?}", e))
+                })?) == 1,
+            ),
+            TypeName::Int16 => {
+                ScalarImpl::Int16(i16::from_be_bytes(b.as_slice().try_into().map_err(
+                    |e| InternalError(format!("Failed to deserialize i16, reason: {:?}", e)),
+                )?))
+            }
+            TypeName::Int32 => {
+                ScalarImpl::Int32(i32::from_be_bytes(b.as_slice().try_into().map_err(
+                    |e| InternalError(format!("Failed to deserialize i32, reason: {:?}", e)),
+                )?))
+            }
+            TypeName::Int64 => {
+                ScalarImpl::Int64(i64::from_be_bytes(b.as_slice().try_into().map_err(
+                    |e| InternalError(format!("Failed to deserialize i64, reason: {:?}", e)),
+                )?))
+            }
+            TypeName::Float => ScalarImpl::Float32(
+                f32::from_be_bytes(b.as_slice().try_into().map_err(|e| {
+                    InternalError(format!("Failed to deserialize f32, reason: {:?}", e))
+                })?)
+                .into(),
+            ),
+            TypeName::Double => ScalarImpl::Float64(
+                f64::from_be_bytes(b.as_slice().try_into().map_err(|e| {
+                    InternalError(format!("Failed to deserialize f64, reason: {:?}", e))
+                })?)
+                .into(),
+            ),
+            TypeName::Varchar => ScalarImpl::Utf8(
+                std::str::from_utf8(b)
+                    .map_err(|e| {
+                        InternalError(format!("Failed to deserialize varchar, reason: {:?}", e))
+                    })?
+                    .to_string(),
+            ),
+            TypeName::Decimal => {
+                ScalarImpl::Decimal(Decimal::from_str(std::str::from_utf8(b).unwrap()).map_err(
+                    |e| InternalError(format!("Failed to deserialize decimal, reason: {:?}", e)),
+                )?)
+            }
+            TypeName::Interval => ScalarImpl::Interval(IntervalUnit::from_protobuf_bytes(
+                b,
+                data_type.get_interval_type()?,
+            )?),
+            TypeName::Struct => {
+                let struct_value: ProstStructValue = Message::decode(b.as_slice())?;
+                let fields: Vec<Datum> = struct_value
+                    .fields
+                    .iter()
+                    .zip_eq(data_type.field_type.iter())
+                    .map(|(b, d)| {
+                        if b.is_empty() {
+                            Ok(None)
```

Comment:
Here makes array/struct with empty string deserialized as NULL

Reply:
Thanks! Would you like to create a issue or PR to fix it?